src/udisksstate.c

```diff
@@ -1794,6 +1794,82 @@ udisks_state_add_loop (UDisksState   *state,
 }
 
 /**
+ * node_cb:
+ * @child:          Element in the list you are visiting
+ * @compare_data:   Data used for comparison
+ * @user_data:      Data supplied by call to iterate list
+ *
+ * Returns: %TRUE if the iteration should stop, else %FALSE
+ */
+typedef gboolean (*node_cb)(GVariant *child, void *compare_data,
+                            void *user_data);
+
+
+/**
+ * interate_list:
+ * @list:           The list to iterate over
+ * @visit:          The function called on each element in the list
+ * @compare_data:   Data used for comparison
+ * @user_data:      Pointer to user supplied data
+ *
+ * Returns: %TRUE if iteration was stopped by node_cb, else %FALSE
+ */
+static gboolean
+interate_list (GVariant *list, node_cb visit, void *compare_data,
+               void *user_data)
+{
+  gboolean rc = FALSE;
+
+  if (list)
```

Comment:
Please do ``if (! list) return FALSE;`` here instead of having almost the whole function under an ``if`` block.